packages/diagram/src/LikeC4Search.tsx

```diff
@@ -1,83 +1,142 @@
   const view = useDiagramState(s => s.view)
   const model = useLikeC4Model(true)
   const store = useDiagramStoreApi()
-  const portalProps = useMantinePortalProps()
+  const [ query, setQuery ] = useState('')
 
-  const getViewActions = (): SpotlightActionGroupData => {
-    const views = model.views()
-    return {
-      group: 'Views',
-      actions: map(views, v => ({
-        id: v.id,
-        label: v.title ?? v.id,
+  const getNodeActionsData = (): SpotlightActionData[] => {
+    const { focusOnNode } = store.getState()
+
+    return pipe(
+      view.nodes,
+      filter(n => !!n.title),
+      map(n => ({
+        id: n.id,
+        label: n.title,
         keywords: [
-          v.id,
-          ...(v.tags ?? []),
-          ...(v.view.description ? [v.view.description] : [])
-        ],
-        onClick: () => {
-          store.setState({
-            hoveredNodeId: null,
-            lastOnNavigate: {
-              fromView: view.id,
-              toView: v.id,
-              fromNode: null
-            }
-          })
-          store.getState().onNavigateTo?.(v.id)
-        },
-        leftSection: <IconSitemap />
+          n.title,
+          ...(n.tags ?? []).map(t => `#${t}`),
+          ...(n.description ? [n.description] : [])
+        ].filter(k => k.toLowerCase()),
+        onClick: () => focusOnNode(n.id),
+        leftSection: <IconRectangularPrism />
       }))
-    }
+    );
   }
 
-  const getNodeActions = (): SpotlightActionGroupData => {
-    return {
-      group: 'Elements',
-      actions: pipe(
-        view.nodes,
-        filter(n => !!n.title),
-        map(n => ({
-          id: n.id,
-          label: n.title,
-          keywords: [
-            n.id,
-            ...(n.tags ?? []),
-            ...(n.description ? [n.description] : [])
-          ],
-          onClick: () => store.getState().focusOnNode(n.id),
-          leftSection: <IconRectangularPrism />
-        }))
-      )
-    }
+  const getViewActionsData = (): SpotlightActionData[] => {
+    const { onNavigateTo } = store.getState()
+    const views = model.views()
+
+    return map(views, v => ({
+      id: v.id,
+      label: v.title ?? v.id,
+      keywords: [
+        v.title ?? v.id,
+        ...(v.tags ?? []),
+        ...(v.view.description ? [v.view.description] : [])
+      ],
+      onClick: () => {
+        store.setState({
+          hoveredNodeId: null,
+          lastOnNavigate: {
+            fromView: view.id,
+            toView: v.id,
+            fromNode: null
+          }
+        })
+        onNavigateTo?.(v.id)
+      },
+      leftSection: <IconSitemap />
+    }));
   }
 
-  const actions: SpotlightActionGroupData[] = useMemo(() => [
-    getNodeActions(),
-    getViewActions()
-  ], [model, store, view])
+  const nodeActions = buildFilteredActions(useMemo(getNodeActionsData, [model, store, view]), query);
+  const viewActions = buildFilteredActions(useMemo(getViewActionsData, [model, store, view]), query);
```

Comment:
What about wrapping in single `useMemo`, like:

```suggestion
  const { nodeActions, viewActions } = useMemo(() => ({
      nodeActions: buildFilteredActions(getNodeActionsData, query),
      viewActions: buildFilteredActions(getViewActionsData, query)
    }),
    [model, store, view, query]
  )
```

Reply:
Changed as requested.